src/kernel/arch/x86_64/gdt.h

```diff
@@ -12,12 +12,14 @@ struct gdt_entry
     u8int base_middle;
     u8int base_high;
 }__attribute__((packed));
```

Comment:
You didn't add `__attribute__((packed))` in `macro.h`

This will allow us to do something like

```c
typedef struct packed
{
   ...
};
```